### ISSUE TITLE
d8 with '--print-all-code' generates segmentation fault

### ISSUE THREAD

**Issue by atiq-cs** [ibmruntimes/v8#48]
`OS/Arch: 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u2 (2017-06-26) x86_64 GNU/Linux`

On this x64 machine, build command lines are as following,

```
tools/dev/v8gen.py ppc64.debug
ninja -C out.gn/ppc64.debug d8
out.gn/ppc64.debug/d8 hello.js --print-all-code
```

It throws following on output,
```
0x209ee7784320   2c0  78a507c6       rldicr  r5, r5, 32, 31
0x209ee7784324   2c4  64a58064       oris    r5, r5, 32868
0x209ee7784328   2c8  60a578c0       ori     r5, r5, 30912
0x209ee778432c   2cc  e8a50000       ld      r5, 0(r5)
0x209ee7784330   2d0  3884005f       addi    r4, r4, 95
0x209ee7784334   2d4  8384ffec       lwz     r28, -20(r4)
0x209ee7784338   2d8  7f9c2214       add     r28, r28, r4
0x209ee778433c   2dc  7d842a14       add     ip, r4, r5
0x209ee7784340   2e0  7d8903a6       mtctr   ip
0x209ee7784344   2e4  4e800420       bctr
Received signal 11 SEGV_MAPERR 00000000000b

==== C stack trace ===============================

 [0x7f0f4442949e]
 [0x7f0f444293f5]
 [0x7f0f440260c0]
 [0x563b7f02ef9c]
 [0x7f0f42c91881]
 [0x7f0f42c9179b]
 [0x7f0f42c8ff4e]
 [0x7f0f434ec5c9]
 [0x7f0f434c316b]
 [0x7f0f42c96371]
 [0x7f0f42c8e32b]
 [0x7f0f434171b4]
 [0x7f0f42cc397c]
 [0x7f0f42ccc755]
 [0x7f0f42ccc0e0]
 [0x7f0f42ccb985]
 [0x7f0f42cc68d8]
 [0x7f0f42cc8c03]
 [0x7f0f4294fbc3]
 [0x7f0f42950196]
 [0x563b7f00c6fd]
 [0x563b7f00c9b5]
 [0x563b7f01c6e4]
 [0x563b7f0200a0]
 [0x563b7f020bbc]
 [0x563b7f020e52]
 [0x7f0f418982b1]
 [0x563b7f0070b7]
[end of stack trace]

```

This issues is observed on following v8 releases,

- 6.2.84 (and probably later)
- 6.2.81
- 6.2.77
- 6.2.66

**Comment by jbajwa** [comment]
Hi,
We did see a similar issue a month ago but was resolved in `6.1.307` commit https://github.com/v8/v8/commit/0216361dc29b6316b9e3347caeffd244a1b47d72 .
Although, this issue seems to be affecting other platforms as well (seg faults on x64 build), but its now resolved in `6.2.92` under commit https://github.com/v8/v8/commit/3c31e109eac88f2ab01d12aeb1cc8e24f7c9b6e9 (I just verified on a ppc64 build).

**Comment by atiq-cs** [comment]
Resolved in release 6.2.92.
